public/scripts/project.js

```diff
@@ -0,0 +1,245 @@
+define(function(require) {
+  var $ = require("jquery");
+
+  var PROJECT_META_KEY = "thimble-project-meta";
+  var Path = Bramble.Filer.Path;
+  var Buffer = Bramble.Filer.Buffer;
+
+  var _host;
+  var _publishUrl;
+
+  var _user;
+  var _id;
+  var _title;
+  var _fs;
+
+  // _dateCreated ?
+  // _dateUpdated? 
+  // _description?
+
+  function getUser() {
+    return _user;
+  }
+
+  function getID() {
+    return _id;
+  }
+
+  function getHost() {
+    return _host;
+  }
+
+  function getPublishUrl() {
+    return _publishUrl;
+  }
+
+  function getRoot() {
+    if(!_user) {
+      return Path.join("/", _projectTitle);
+    }
+
+    return Path.join("/", _user, "projects", _id);
+  }
+
+  // From /7/projects/5/index.html to /index.html
+  function stripRoot(path) {
+    return path.replace(this.getRoot(), "");
+  }
+  
+  // From /index.html to /7/projects/5/index.html to
+  function addRoot(path) {
+    Path.join(getRoot(), path);
+  }
+
+  // Installs a tarball (arraybuffer) containing the project's files/folders.
+  function installTarball(tarball, callback) {
+    var untarWorker;
+    var pending = null;
+    var root = getRoot();
+    var sh = new _fs.Shell();
+
+    function extract(path, data, callback) {
+      path = Path.resolve(root, path);
```

Comment:
This won't work. `Path.resolve("/1/projects/5", "/index.html")` results in `/index.html`. Instead, we should probably just follow the convention that we are sending `/index.html` and instead just call `addRoot(path)`.